react-native-purchases-ui/ios/RCPaywallFooterViewManager.m

```diff
@@ -0,0 +1,144 @@
+//
+//  RCPaywallFooterViewManager.m
+//  RNPaywalls
+//
+//  Created by Cesar de la Vega on 29/12/23.
+//
+
+#import "RNPaywalls.h"
+#import "RCPaywallFooterViewManager.h"
+
+@import RevenueCatUI;
+@import PurchasesHybridCommon;
+
+#import "UIView+Extensions.h"
+
+#import <React/RCTShadowView.h>
+#import <React/RCTUIManager.h>
+#import <React/RCTBridge.h>
+#import <React/RCTRootViewDelegate.h>
+#import <React/RCTEventEmitter.h>
+
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface FooterViewWrapper : UIView
+
+- (instancetype)initWithFooterViewController:(UIViewController *)footerViewController 
+                                      bridge:(RCTBridge *)bridge;
+
+@end
+
+NS_ASSUME_NONNULL_END
+
+@interface FooterViewWrapper () <RCPaywallViewControllerDelegate>
+
+@property (strong, nonatomic) UIViewController *footerViewController;
+@property (strong, nonatomic) RCTBridge *bridge;
+
+@property BOOL addedToHierarchy;
+
+@end
+
+@implementation FooterViewWrapper
+
+- (instancetype)initWithFooterViewController:(UIViewController *)footerViewController bridge:(RCTBridge *)bridge {
+    if ((self = [super initWithFrame:footerViewController.view.bounds])) {
+        _bridge = bridge;
+        _footerViewController = footerViewController;
+    }
+
+    return self;
+}
+
+- (void)safeAreaInsetsDidChange {
+    [super safeAreaInsetsDidChange];
+
+    // Get the safe area insets, for example
+    UIEdgeInsets safeAreaInsets = self.safeAreaInsets;
+
+//    TODO: figure out a better way of sending event, since this is deprecated
+//    It's probably better to create a singleton from the module that this view manager can call and use to send events
```

Comment:
do we have a task or something scheduled to come back to this one?

Reply:
We decided it wasn't worth spending a lot of time on this right now, but should do it in case React Native ends up removing this deprecated method.